From 9bd96cb759e39912ad237e140be3548d0397442e Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Mon, 14 Aug 2017 10:56:08 +0200
Subject: [PATCH 4/8] lib/randutils: reset lose counter

It's probaly good idea to reset lose counter when we fallback from
getrandom() to /dev/urandom.

Signed-off-by: Karel Zak <kzak@redhat.com>
---
 lib/randutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/randutils.c b/lib/randutils.c
index 5cd1c99..43afc60 100644
--- a/lib/randutils.c
+++ b/lib/randutils.c
@@ -120,7 +120,6 @@ void random_get_bytes(void *buf, size_t nbytes)
 
 	if (errno == ENOSYS)
 #endif
-
 	/*
 	 * We've been built against headers that support getrandom, but the
 	 * running kernel does not.  Fallback to reading from /dev/{u,}random
@@ -129,6 +128,7 @@ void random_get_bytes(void *buf, size_t nbytes)
 	{
 		int fd = random_get_fd();
 
+		lose_counter = 0;
 		if (fd >= 0) {
 			while (n > 0) {
 				ssize_t x = read(fd, cp, n);
-- 
2.7.4

